operator-framework/src/main/java/io/javaoperatorsdk/operator/ControllerUtils.java

```diff
@@ -1,89 +1,39 @@
 package io.javaoperatorsdk.operator;
 
+import io.fabric8.kubernetes.client.CustomResource;
 import io.javaoperatorsdk.operator.api.Controller;
 import io.javaoperatorsdk.operator.api.ResourceController;
-import io.fabric8.kubernetes.api.builder.Function;
-import io.fabric8.kubernetes.client.CustomResource;
-import io.fabric8.kubernetes.client.CustomResourceDoneable;
-import javassist.*;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import java.util.HashMap;
-import java.util.Map;
 
 
 public class ControllerUtils {
-
-    private final static double JAVA_VERSION = Double.parseDouble(System.getProperty("java.specification.version"));
-    private static final String FINALIZER_NAME_SUFFIX = "/finalizer";
-
-    // this is just to support testing, this way we don't try to create class multiple times in memory with same name.
-    // note that other solution is to add a random string to doneable class name
-    private static Map<Class<? extends CustomResource>, Class<? extends CustomResourceDoneable<? extends CustomResource>>>
-            doneableClassCache = new HashMap<>();
-
+    private static final String FINALIZER_NAME_SUFFIX = ".javaoperatorsdk.io/finalizer";
+    
     static String getFinalizer(ResourceController controller) {
         final String annotationFinalizerName = getAnnotation(controller).finalizerName();
         if (!Controller.NULL.equals(annotationFinalizerName)) {
             return annotationFinalizerName;
         }
-        final String crdName = getAnnotation(controller).crdName() + FINALIZER_NAME_SUFFIX;
-        return crdName;
+        return getDefaultFinalizerIdentifier(controller);
     }
-
+    
+    static String getDefaultFinalizerIdentifier(ResourceController controller) {
+        final Class<? extends ResourceController> controllerClass = controller.getClass();
+        return controllerClass.getSimpleName().toLowerCase() + FINALIZER_NAME_SUFFIX;
```

Comment:
The class name can be changed due to refactoring, it's risky to use it as part of the finalizer name. i.e Renaming the controller class name without changing the CR name shouldn't need data migration.

Reply:
This is default fallback if there is no name defined, so even after refactoring developers can introduce own name. I would keep it like that.

Reply:
The developer can easily overlook this fact that just because a simple refactoring in the class name they now have to explicitly set the finalized name. 
What do you think is wrong with the current implementation? (`crdName`/`finalizer`)

Reply:
Ah, I got your point. Then it would be nice to use crd name and revert change pf `FINALIZER_NAME_SUFFIX` const

Reply:
The current implementation of the `getFinalizer` (in the master) does so.

Reply:
The issue I have with reverting this change is that, currently, the CR names we have don't really follow the spec for finalizer names… so we're not really setting a good example, which I think we should. Are we OK with renaming the CRDs so that their names provide good examples?

Reply:
Yes, definitely we should update examples to provided best practices.

Reply:
> Are we OK with renaming the CRDs so that their names provide good examples?
🙌